apps/np-atobarai/src/app/api/webhooks/saleor/transaction-refund-requested/refund-strategy/after-fulfillment-strategies.ts

```diff
@@ -0,0 +1,377 @@
+import {
+  PartialRefundWithLineItemsGoodsBuilder,
+  PartialRefundWithoutLineItemsGoodsBuilder,
+} from "@/modules/atobarai/atobarai-goods/refund-goods-builders";
+import { createAtobaraiMoney } from "@/modules/atobarai/atobarai-money";
+import { createAtobaraiShopOrderDate } from "@/modules/atobarai/atobarai-shop-order-date";
+import {
+  AtobaraiTransactionId,
+  createAtobaraiTransactionId,
+} from "@/modules/atobarai/atobarai-transaction-id";
+import { createSaleorTransactionToken } from "@/modules/saleor/saleor-transaction-token";
+import {
+  RefundFailureResult,
+  RefundSuccessResult,
+} from "@/modules/transaction-result/refund-result";
+
+import { MalformedRequestResponse } from "../../saleor-webhook-responses";
+import { TransactionRefundRequestedUseCaseResponse } from "../use-case-response";
+import { AfterFulfillmentRefundContext, RefundStrategy } from "./types";
+
+export class AfterFulfillmentFullRefundStrategy implements RefundStrategy {
+  private readonly logger = createLogger("AfterFulfillmentFullRefundStrategy");
+
+  async execute(
+    context: AfterFulfillmentRefundContext,
+  ): Promise<Result<TransactionRefundRequestedUseCaseResponse, MalformedRequestResponse>> {
+    const { atobaraiTransactionId, apiClient } = context;
+
+    const payload = createAtobaraiCancelTransactionPayload({
+      atobaraiTransactionId,
+    });
+
+    const cancelResult = await apiClient.cancelTransaction(payload, {
+      checkForMultipleResults: true,
+    });
+
+    if (cancelResult.isErr()) {
+      this.logger.error("Failed to cancel Atobarai transaction", {
+        error: cancelResult.error,
+      });
+
+      return ok(
+        new TransactionRefundRequestedUseCaseResponse.Failure({
+          transactionResult: new RefundFailureResult(),
+        }),
+      );
+    }
+
+    return ok(
+      new TransactionRefundRequestedUseCaseResponse.Success({
+        transactionResult: new RefundSuccessResult(),
+        atobaraiTransactionId,
+      }),
+    );
+  }
+}
+
+export class AfterFulfillmentPartialRefundWithLineItemsStrategy implements RefundStrategy {
+  private readonly logger = createLogger("AfterFulfillmentPartialRefundWithLineItemsStrategy");
+  private readonly goodsBuilder = new PartialRefundWithLineItemsGoodsBuilder();
+
+  async execute(
+    context: AfterFulfillmentRefundContext,
+  ): Promise<Result<TransactionRefundRequestedUseCaseResponse, MalformedRequestResponse>> {
+    const {
+      parsedEvent,
+      appConfig,
+      atobaraiTransactionId,
+      apiClient,
+      trackingNumber,
+      shippingCompanyCode,
+    } = context;
+
+    const cancelTransactionResult = await this.cancelTransaction(atobaraiTransactionId, apiClient);
+
+    if (cancelTransactionResult.isErr()) {
+      return ok(
+        new TransactionRefundRequestedUseCaseResponse.Failure({
+          transactionResult: new RefundFailureResult(),
+        }),
+      );
+    }
```

Comment:
Let's pass debug messuge to result, so the merchant can recover from this situation. Incldue reason and notice to go to atobarai dashboard to fix that